nix/shell.nix

```
@@ -0,0 +1,31 @@
+{
+  lib,
+  stdenv,
+  mkShell,
+  steel,
+  cargo,
+  rustc,
+  openssl,
+  libiconv,
+  CoreServices,
+  SystemConfiguration,
+  pkg-config,
+  rust-analyzer,
+  rustfmt,
+}:
+mkShell {
+  shellHook = ''
+    export STEEL_HOME="${steel}/lib/"
+  '';
+  buildInputs =
+    [cargo rustc openssl libiconv]
+    ++ lib.optionals stdenv.isDarwin [
+      CoreServices
+      SystemConfiguration
+    ];
+  nativeBuildInputs = [
+    pkg-config
+    rust-analyzer
+    rustfmt
+  ];
```

```suggestion
  packages = [
    cargo
    rustc
    rust-analyzer
    rustfmt
    libiconv
  ] ++ lib.optionals stdenv.isDarwin [
    CoreServices
    SystemConfiguration
  ];
  
  inputsFrom = steel;
```

`packages` should be used in shells, as the logical difference between `nativeBuildInputs` and `buildInputs` no longer applies.

`inputsFrom` takes pre-existing `nativeBuildInputs` and `buildInputs` from the given package and applies them to the shell.